src/lib.rs

```diff
@@ -523,23 +552,8 @@ impl Socket {
         Ok(parts)
     }
 
-    pub fn close(&mut self) -> Result<()> {
-        if !self.closed {
-            self.closed = true;
-
-            if unsafe { zmq_sys::zmq_close(self.sock) } == -1i32 {
-                return Err(errno_to_error());
-            }
-        }
-
-        Ok(())
-    }
-
-    pub fn close_final(&mut self) -> Result<()> {
-        if !self.closed && unsafe { zmq_sys::zmq_close(self.sock) } == -1i32 {
-            return Err(errno_to_error());
-        }
-
+    #[deprecated(note="`close` is handled implicitly by the Drop trait")]
+    pub fn close(self) -> Result<()> {
```

Comment:
I think it's fine to just remove the close method. We are going to be doing some breaking API changes anyway.